src/components/pages/ProblemDetailedPage/index.tsx

```diff
@@ -1,54 +1,127 @@
 import { FC, useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import { useQuery } from "@apollo/client";
+import { format } from "date-fns";
+import { pl } from "date-fns/locale";
 
-import { GetProblemsQuery, PROBLEMS } from "../ProblemsPage/interface";
 import ProblemSubtitle from "../../atoms/ProblemSubtitle";
-import Wrapper from "../../atoms/Wrapper";
-import { white, secondarySubtitle } from "../../../assets/styles/colors";
+import Tag from "../../atoms/Tag";
+import ScrollToTopButton from "../../atoms/ScrollToTopButton";
+import Button from "../../atoms/Button";
+import { GetProblemDetailsQuery, PROBLEM_DETAILS } from "./interface";
+import { secondarySubtitle } from "../../../assets/styles/colors";
+import * as S from "./styles";
 
 const ProblemDetailedPage: FC = () => {
   const params = useParams();
   const id: string = params.problemId || "";
 
-  const { data, loading } = useQuery<GetProblemsQuery>(PROBLEMS);
-  const [problemDetails, setProblemDetails] = useState({
-    title: "",
-    author: "",
+  const { data, loading } = useQuery<GetProblemDetailsQuery>(PROBLEM_DETAILS, {
+    variables: { id },
   });
+  const [problemDetails, setProblemDetails] = useState<any>({});
 
-  const problemsLoaded = data?.problems && !loading;
+  const problemsLoaded = data && !loading;
 
   useEffect(() => {
     if (problemsLoaded) {
-      const { problems } = data;
-
-      const currentProblem = problems[parseInt(id) - 1];
-
+      const { details } = data.problem;
       setProblemDetails({
-        title: currentProblem.title,
-        author: currentProblem.author,
+        ...details,
       });
     }
-  }, [data?.problems.length, loading]);
+  }, [data, loading]);
+
+  const formattedDate =
+    problemDetails.createdAt &&
+    format(new Date(problemDetails.createdAt), "d MMM y", {
+      locale: pl,
+    });
```

Comment:
We could extract it and make it a function in some `utils.ts` file.

Reply:
Done ;)